modules/core/src/main/java/org/apache/synapse/config/xml/endpoints/EndpointDefinitionFactory.java

```diff
@@ -159,36 +162,27 @@ public EndpointDefinition createDefinition(OMElement elem) {
                 String d = duration.getText();
                 if (d != null) {
                     try {
-                        Pattern pattern = Pattern.compile("\\{.*\\}");
-                        if (pattern.matcher(d).matches()) {
-                            d = d.trim().substring(1, d.length() - 1);
-                            SynapseXPath xpath = new SynapseXPath(d);
-                            definition.setDynamicTimeoutExpression(xpath);
-                        } else {
-                            long timeoutMilliSeconds = Long.parseLong(d.trim());
-                            definition.setTimeoutDuration(timeoutMilliSeconds);
+                        Value timeoutDurationValue = valueFactory.createTextValue(duration);
+                        if (timeoutDurationValue.getKeyValue() != null) {
+                            Long.parseLong(timeoutDurationValue.getKeyValue());
                         }
+                        definition.setTimeoutDuration(valueFactory.createTextValue(duration));
                     } catch (NumberFormatException e) {
                         handleException("Endpoint timeout duration expected as a " +
                                 "number but was not a number");
-                    } catch (JaxenException e) {
-                        handleException("Couldn't assign dynamic endpoint timeout as Synapse expression");
                     }
                 }
             }
 
             OMElement action = timeout.getFirstChildWithName(
                     new QName(XMLConfigConstants.SYNAPSE_NAMESPACE, "responseAction"));
             if (action != null && action.getText() != null) {
-                String actionString = action.getText();
-                if ("discard".equalsIgnoreCase(actionString.trim())) {
-                    definition.setTimeoutAction(SynapseConstants.DISCARD);
-                } else if ("fault".equalsIgnoreCase(actionString.trim())) {
-                    definition.setTimeoutAction(SynapseConstants.DISCARD_AND_FAULT);
-                } else {
-                    handleException("Invalid timeout action, action : "
-                            + actionString + " is not supported");
+                Value timeoutActionValue = valueFactory.createTextValue(action);
+
+                if (timeoutActionValue.getKeyValue() != null && !timeoutActionValue.getKeyValue().equals("discard") && !timeoutActionValue.getKeyValue().equals("fault")) {
```

Comment:
Shall we check the formatting here?

Reply:
Also shall we use a constant for "discard"?

Reply:
added